doc/dev/crimson/poseidonstore.rst

```diff
@@ -0,0 +1,470 @@
+            v                                |                   |   
+       +---------+---------+---------+       |                   |     
+       |          Radix tree         |       |                   v     
+       +=========+=========+=========+       |      +---------------+       
+       | onode   | ...     | ...     |       |      |               |
+       +---------+---------+---------+       |      | Num Chunk     |        
+    +--| onode   | ...     | ...     |       |      | <Offset, len> |        
+    |  +---------+---------+---------+       |      | <Offset, len> |-------|
+    |                                        |      | ...           |       | 
+    |                                        |      +---------------+       | 
+    |                                        |      ^                       |
+    |                                        |      |                       |
+    |                                        |      |                       |   
+    |                                        |      |                       |   
+    |  +---------------+  +-------------+    |      |                       v   
+    +->| onode         |  | onode       |<---|      |       +------------+------------+  
+       +===============+  +=============+           |       | Block0     | Block1     |  
+       | OID           |  | OID         |           |       +============+============+  
+       | Omaps         |  | Omaps       |           |       | Data       | Data       |  
+       | Data Extent   |  | Data Extent |-----------|       +-------------------------+
+       +---------------+  +-------------+           
+
+WAL
+---
+Each SP has a WAL.
+The datas written to the WAL are metadata updates, free space update and small data.
+Note that only data smaller than the predefined threshold needs to be written to the WAL.
+The larger data is written to the unallocated free space and its onode's extent_map is updated accordingly 
+(also on-disk extent map). We statically allocate WAL partition aside from data partition pre-configured.
+
+
+Partition and Reactor thread
+----------------------------
+In early stage development, PoseidonStore will employ static allocation of partition. The number of sharded partitions
+is fixed and the size of each partition also should be configured before running cluster.
+But, the number of partitions can grow as below. We leave this as a future work.
+   +------+------+-------------+----------------+
+   | SP 1 | SP N | -->     <-- | gobal partition|
+   +------+------+-------------+----------------+ 
+Also, each reactor thread has a static set of SPs.
+
+
+Cache
+-----
+There are mainly two cache data structures; onode cache and block cache.
+It looks like below.
+
+1.Onode cache:
+lru_map <OID, OnodeRef>;
+Onode {
+  extent tree block_maps;
+  omap lists omaps;
+}
+2. Block cache:
+Omap cache --> lru_map <OID, <omap_key, value>>
```

Comment:
I think you'd be better off caching omap via a more general block cache.  Explicitly representing each key seems complicated and cpu expensive.